cloudslang_dsl_reference.rst

```diff
@@ -101,6 +102,82 @@ and concepts are explained in detail below.
 -  `outputs <#outputs>`__
 -  `results <#results>`__
 
+.. _expressions:
+
+Expressions
+===========
+
+Many CloudSlang keys map to either an expression or literal value.
+
+Literal Values
+--------------
+
+Literal values are denoted as they are in standard YAML. Numbers are interpreted
+as numerical values and strings may be written unquoted, single quoted or double
+quoted.
+
+**Example: literal values**
+
+.. code-block:: yaml
+
+    literal_number: 4
+    literal_unquoted_string: cloudslang
+    literal_single_quoted_string: 'cloudslang'
+    literal_double_quoted_string: "cloudslang"
+
+**Note:** The use of unquoted strings where an expression is allowed is strongly
+discouraged.
+
+Standard Expressions
+--------------------
+
+Expressions are preceded by a dollar sign (``$``) and enclosed in curly brackets
+(``{}``).
+
+**Example: expressions**
+
+.. code-block:: yaml
+
+    - expression_1: ${4 + 7}
+    - expression_2: ${some_input}
+    - expression_3: ${get('input1', 'default_input')}
+
+Expressions with Special Characters
+-----------------------------------
+
+Expressions that contain characters that are considered special characters in
+YAML must be enclosed in quotes or use YAML block notation. If using quotes, use
+the style of quotes that are not already used in the expression. For example, if
+your expression contains single quotes (``'``), enclose the expression using
+double quotes (``"``).
+
+**Example: escaping special characters**
+
+.. code-block:: yaml
+
+    - expression1: "${var1 + ': ' + var2}"
+    - expression2: >
+        ${var1 + ': ' + var2}
+    - expression3: |
+        ${var1 + ': ' + var2}
+
+Maps
+----
+
+To pass a map where an expression is allowed use the `default <#default>`__
+property. It is possible to use two sets of quotes and an expression marker
+instead, but that approach is not as clean.
```

Comment:
I think that it is not clear enough what you mean by this sentence. I think it is better to divide the examples below to 2 sections. Give the preferred syntax examples after the first sentence, and he unfavorable syntax examples after the second sentence